homeassistant/components/glances/sensor.py

```diff
@@ -80,6 +81,11 @@ def __init__(
 
         self.entity_description = description
         self._attr_name = f"{name} {sensor_name_prefix} {description.name_suffix}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, glances_data.host)},
```

Comment:
I am not an expert on this, but i read hostname as not a source for unique id https://developers.home-assistant.io/docs/entity_registry_index?_highlight=unique#unacceptable-sources-for-a-unique-id ?

Just flagging, bug will defer to others

Reply:
This is not a unique_id (which by the way actually uses some problematic sources in this integration).
This is a device identifier - basically, a lookup key for entities that should belong to device. I’m not sure this is as strict as unique_id but open to suggestions for better sources.

Reply:
Looks good to me

Reply:
It's the host of the Glances server that you enter when setting up the integration. Essentially, an IP.
I'm not sure if Glances provides a mac address we could use and I think host is good enough to uniquely identify a Glances server.

Reply:
> Essentially, an IP.

In that case, Allen was correct. IP is not allowed, as a matter of fact, looking at the existing code, this integration already violates that rule for its entities big time.

It uses the "name" in its unique ID for its entities (which is an absolute no-no!).

Reply:
Well, I know that name is a big no-no and consider fixing it somehow even though it's probably very tricky without big breaking changes...
Do I understand correctly that we should use entry_id here?

Reply:
I don't see any unique identifier for the machine hosting glances that could be used here. Unless you can find something @avee87 I recommend to use the config entry_id.

Reply:
> Well, I know that name is a big no-no and consider fixing it somehow even though it's probably very tricky without big breaking changes...

We have migrations helpers in place to migrate entries on all places. It happens from time to time and can all be done without causing breaking changes